tests/rados/test_stretch_site_down.py

```diff
@@ -0,0 +1,289 @@
+        )
+
+    # perform rados put to check if write ops is possible
+    pool_obj.do_rados_put(client=client_node, pool=pool_name, nobj=200, timeout=50)
+
+    log.debug("sleeping for 20 seconds for the objects to be displayed in ceph df")
+    time.sleep(20)
+
+    pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+    log.debug(pool_stat)
+
+    # Objects should be more than the initial no of objects
+    if pool_stat["stats"]["objects"] <= init_objects:
+        log.error(
+            "Write ops should be possible, number of objects in the pool has not changed"
+        )
+        raise Exception(f"Pool {pool_name} has {pool_stat['stats']['objects']} objs")
+    log.info(
+        f"Successfully wrote {pool_stat['stats']['objects']} on pool {pool_name} in degraded mode\n"
+        f"Proceeding to bring up the nodes and recover the cluster from degraded mode"
+    )
+
+    if shutdown_site in [dc_1_name, dc_2_name]:
+        log.debug(f"Proceeding to reboot data site {dc_1_name}")
+        for host in dc_1_hosts:
+            log.debug(f"Proceeding to reboot host {host}")
+            if not host_restart(gyaml=osp_cred, name=host):
+                log.error(f"Failed to restart host : {host}")
+                raise Exception("Test execution Failed")
+        log.info(
+            f"Completed restart of all the hosts in site {dc_1_name}. Host names : {dc_1_hosts}"
+        )
+
+        # Sleeping for 20 seconds for the hosts to be added back to cluster
+        time.sleep(20)
+
+        # Waiting for Cluster to be active + Clean post boot of all hosts
+        if not wait_for_clean_pg_sets(rados_obj):
+            log.error(
+                "Cluster did not reach active + Clean state post restart of failed DC"
+            )
+            raise Exception("Stretch mode site down tests Failed")
+
+        # Checking health of stretch mode on the cluster. Should not be degraded any longer
+        stretch_details = rados_obj.get_stretch_mode_dump()
+        if stretch_details["degraded_stretch_mode"]:
+            log.error(
+                f"Stretch Cluster is marked as degraded even though all sites are up and healthy : {stretch_details}"
+            )
+            raise Exception("Stretch mode degraded test Failed on the provided cluster")
+        log.info("Degraded Stretch Cluster mode exited post active + Clean. Pass")
+
+    else:
+        log.info("Restarting arbiter mon site")
+        for host in tiebreaker_hosts:
+            log.debug(f"Proceeding to Restart host {host}")
+            if not host_restart(gyaml=osp_cred, name=host):
+                log.error(f"Failed to Restart host : {host}")
+                raise Exception("Test execution Failed")
+        time.sleep(20)
+        log.info(
+            f"Completed Restart of all the hosts in site {shutdown_site}. Host names : {tiebreaker_hosts}"
+        )
+        time.sleep(20)
+
```

Comment:
Are the active + Clean checks and cluster health check done for data site restarted scenario needs be checked here as well.

Reply:
Added common checks for both the scenarios.